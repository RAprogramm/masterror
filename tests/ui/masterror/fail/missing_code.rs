// SPDX-FileCopyrightText: 2025 RAprogramm <andrey.rozanov.vl@gmail.com>
//
// SPDX-License-Identifier: MIT

use masterror::{AppErrorKind, Masterror};

#[derive(Debug, Masterror)]
#[error("oops")]
#[masterror(category = AppErrorKind::Internal)]
struct MissingCode;

fn main() {}
