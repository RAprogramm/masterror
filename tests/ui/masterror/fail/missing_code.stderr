error: missing `code = ...` in #[masterror(...)]
 --> tests/ui/masterror/fail/missing_code.rs:5:1
  |
5 | #[masterror(category = AppErrorKind::Internal)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused import: `AppErrorKind`
 --> tests/ui/masterror/fail/missing_code.rs:1:17
  |
1 | use masterror::{AppErrorKind, Masterror};
  |                 ^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` (part of `#[warn(unused)]`) on by default
